iTunesUDataPurge/RemoveITunesUCourses.java

```diff
@@ -478,5 +501,186 @@ public static void fileUrl(String fAddress, String localFileName, String destina
 			}
 		}
 	}
+
+	public static void deletePage(String searchServer, String siteId, String username, String password) throws Exception{
+
+		String pageToDelete = "iTunes U";
+
+		// Create SOAP Connection
+		SOAPConnectionFactory soapConnectionFactory = SOAPConnectionFactory.newInstance();
+		SOAPConnection soapConnection = soapConnectionFactory.createConnection();
+
+		// Send SOAP Message to SOAP Server
+		String sakaiLoginUrl = searchServer + "sakai-axis/SakaiLogin.jws?wsdl";
+		String sakaiScriptUrl = searchServer + "sakai-axis/SakaiScript.jws?wsdl";
+
+		//Making the createLoginRequest call. Returns a response message containing the session id that should be active on server.
+		SOAPMessage soapResponse = soapConnection.call(createLoginRequest(searchServer, username, password), sakaiLoginUrl);
```

Comment:
so a new login request is issued for every iTunesU page delete, right. I did not see any logout call after the delete. It would be tidy to log out those sessions.

I wonder maybe the better way is to use one session for all deletes. We will verify the session (createCheckSessionRequest)  upon every page removal. If it is no longer valid, we create a new session